### PR TITLE
Adding JavaScript functions to sample apps

### PR DESCRIPTION
**Pull request by nickwesselman**
Existing sample apps can add JavaScript functions in this PR. It should not be merged until JavaScript functions are generally available.

* Vanilla JavaScript functions should be added side-by-side with existing Rust functions to all sample apps.
* The function logic should be the same in JS as it is in Rust.
* Vitest should be used to exhibit unit testing.
* Test cases should be the same in JS as they are in Rust.

## 🎩

* Execution of JS functions requires the checkout extensibility developer preview.

### Payment Customization
```
yarn create @shopify/app --template https://github.com/Shopify/function-examples/sample-apps/payment-customizations#sample-apps-javascript
```

### Delivery Customization
```
yarn create @shopify/app --template https://github.com/Shopify/function-examples/sample-apps/delivery-customizations#sample-apps-javascript
```

### Discounts
```
yarn create @shopify/app --template https://github.com/Shopify/function-examples/sample-apps/discounts#sample-apps-javascript
```

### CODE REVIEW

File: sample-apps/payment-customizations/extensions/payment-customization-js/src/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  );
+
+  const cartTotal = parseFloat(input.cart.cost.totalAmount.amount ?? "0.0");
+  if (cartTotal < configuration.cartTotal) {
+    console.error("Cart total is not high enough, no need to hide the payment method.");
+    return NO_CHANGES;
+  }
+
+  const hidePaymentMethod = input.paymentMethods
+    .find(method => method.name.indexOf(configuration.paymentMethodName) != -1);
```
Comment (andrewhassan) [inline sample-apps/payment-customizations/extensions/payment-customization-js/src/index.js:38]:
```suggestion
    .find(method => method.name.includes(configuration.paymentMethodName));
```

**Review by andrewhassan — COMMENTED** [review]
Out of curiosity, why did we choose to have a JS example and not a TS example instead?

File: sample-apps/payment-customizations/extensions/payment-customization-js/package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "payment-customization-js",
+  "version": "0.0.1",
+  "license": "UNLICENSED",
+  "scripts": {
+    "shopify": "npm exec -- shopify",
+    "typegen": "npm exec -- shopify app function typegen",
+    "build": "npm exec -- shopify app function build",
+    "preview": "npm exec -- shopify app function run"
```
[the remark this answers is not in the document]
Comment (nickwesselman) [reply sample-apps/payment-customizations/extensions/payment-customization-js/package.json:9]: There's some reason this is needed, @Arkham would have details. It's in our templates.

**Comment by nickwesselman** [comment]
> Out of curiosity, why did we choose to have a JS example and not a TS example instead?

@andrewhassan Our docs, and thus the sample, are going to focus on JS based on consistency with UI extensions, and survey feedback (Vanilla JS edging out TS). Thought about doing both but it would add maintenance overhead for minor benefit.

**Comment by nickwesselman** [comment]
@adampetro Looks like this is failing now due to Rust checks being done on a JavaScript extension, is that something that was added in the JS unit test PR?

**Comment by nickwesselman** [comment]
@adampetro Thanks! Unfortunately they [can't be globbed](https://github.com/Shopify/function-examples/pull/169/commits/b88a5cae1a476d8158e7e1becbf7584c2b844170) due to [this](https://github.com/rust-lang/cargo/issues/11405) but that fixed it.

Also added the JS sample extensions to the workspace so those tests run now too. Awesome work setting up all the test automation.

**Pull request merged**